src/health_misinfo_shared/app.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any, Iterable
+from flask import Flask, request, jsonify
+from flask.typing import ResponseReturnValue
+import sqlite3
+from sqlite3 import Connection, Row
+
+app = Flask(__name__)
+
+
+def get_db_connection() -> Connection:
+    """row_factory = Row makes the return value a mapping."""
+    conn = sqlite3.connect("database.db")
+    conn.row_factory = Row
+    return conn
+
+
+def execute_sql(sql: str, params: tuple[Any, ...] = ()) -> list[Row]:
+    """Open/close the database for each request, we don't expect to make many."""
+    conn = get_db_connection()
+    cur = conn.cursor()
+    cur.execute("PRAGMA foreign_keys = ON")
+    cur.execute(sql, params)
+    data = cur.fetchall()
+    conn.commit()
+    conn.close()
+    return data
+
+
+@app.get("/transcripts")
+def get_transcripts() -> ResponseReturnValue:
+    transcripts = execute_sql("SELECT * FROM video_transcripts")
+    return jsonify([{**t} for t in transcripts]), 200
+
+
+@app.post("/transcripts")
+def post_transcripts() -> ResponseReturnValue:
+    data = request.get_json()
+    execute_sql(
+        "INSERT INTO video_transcripts (id, url, metadata, transcript) VALUES (?, ?, ?, ?)",
+        (
+            data["id"],
+            data["url"],
+            data["metadata"],
+            data["transcript"],
+        ),
+    )
+    return jsonify(id=data["id"]), 201
+
+
+@app.get("/transcripts/<string:id>")
+def get_transcript(id: str) -> ResponseReturnValue:
+    transcripts = execute_sql("SELECT * FROM video_transcripts WHERE id = ?", (id,))
+    if transcripts:
+        return jsonify(**transcripts[0]), 200
+    return jsonify({"error": "transcript not found"}), 404
+
+
+@app.delete("/transcripts/<string:id>")
+def delete_transcript(id: str) -> ResponseReturnValue:
+    execute_sql("DELETE FROM video_transcripts WHERE id = ?", (id,))
+    execute_sql("DELETE FROM training_claims WHERE video_id = ?", (id,))
+    execute_sql("DELETE FROM inferred_claims WHERE video_id = ?", (id,))
+    return "", 204
```

Comment:
Does this work in SQLite? I don't think it would in e.g. Postgres due to referential integrity (deleting the row in `video_transcripts` that rows in `training_claims` and `inferred_claims` still refer to). 

One solution is to change the order so that rows in `video_transcripts` are deleted last. Another is to set `ON DELETE CASCADE` on the table definitions for `inferred_claims` and `training_claims` which means that when `video_transcripts` has rows deleted, any rows that reference those rows are also deleted in other tables.